### ISSUE TITLE
Update `ListContainersParam` filters

### ISSUE THREAD

**Issue by johnflavin** [spotify/docker-client#710]
There are lots of new filters available when listing containers. From the [latest swagger.yaml](https://github.com/docker/docker/blob/master/api/swagger.yaml#L2624):

> Filters to process on the container list, encoded as JSON (a `map[string][]string`). For example, `{"status": ["paused"]}` will only return paused containers.
> 
> - `ancestor`=(`<image-name>[:<tag>]`, `<image id>`, or `<image@digest>`)
> - `before`=(`<container id>` or `<container name>`)
> - `expose`=(`<port>[/<proto>]`|`<startport-endport>/[<proto>]`)
> - `exited=<int>` containers with exit code of `<int>`
> - `health`=(`starting`|`healthy`|`unhealthy`|`none`)
> - `id=<ID>` a container's ID
> - `isolation=`(`default`|`process`|`hyperv`) (Windows daemon only)
> - `is-task=`(`true`|`false`)
> - `label=key` or `label="key=value"` of a container label
> - `name=<name>` a container's name
> - `network`=(`<network id>` or `<network name>`)
> - `publish`=(`<port>[/<proto>]`|`<startport-endport>/[<proto>]`)
> - `since`=(`<container id>` or `<container name>`)
> - `status=`(`created`|`restarting`|`running`|`removing`|`paused`|`exited`|`dead`)
> - `volume`=(`<volume name>` or `<mount point destination>`)

We should add these new filters. Might also be nice to dig through the older docker API documentation, figure out when each filter was added, annotate the methods with javadoc `@since (docker api version)`.

I think this should wait on #707, which refactors all the filter params.

**Comment by albers** [comment]
I also noticed some missing filters and was thinking about adding them.
I'd like to take over that task.

**Comment by davidxia** [comment]
@johnflavin We can close this as #707 was merged, right?

**Comment by johnflavin** [comment]
No, this is a different issue. #707 refactored all the existing filters to be a consistent style; its only additions were filters for networks. This issue is about adding new filters for containers.

**Comment by davidxia** [comment]
Ah I see. PRs are welcome then :)

**Comment by dmandalidis** [comment]
Hi @albers ,

Since this project went on mature status, please re-open this issue (if it still stands) to https://github.com/dmandalidis/docker-client. Thanks
